docs/spark/tutorials/ml-sentment-analysis.md

```diff
@@ -0,0 +1,207 @@
+
+1. Choose *Sentiment* from the **Columns to Predict** dropdown.
+
+1. On the **Train** page, set the time to train to *60 seconds* and select **Start training**. Notice the status of your training under **Progress**.
+
+1. Once Model Builder is finished training, **Evaluate** the training results. You can type phrases into the text box below **Try your model** and select **Predict** to see the output.
+
+1. Select **Code** and then select **Add Projects** to add the ML model to the solution.
+
+1. Notice that two projects are added to your solutions: **MLSparkModelML.ConsoleApp** and **MLSparkModelML.Model**.
+
+1. Double-click on your *MLSpark* C# project and notice that the following project reference has been added.
+
+   ```xml
+   <ItemGroup>
+       <ProjectReference Include="..\MLSparkModelML.Model\MLSparkModelML.Model.csproj" />
+   </ItemGroup>
+   ```
+
+## Create a console app
+
+Model Builder creates a console app for you.
+
+1. Right-click on **MLSparkModelML.Console** in Solution Explorer, and select **Manage NuGet Packages**.
+
+1. Search for **Microsoft.Spark** and install the package. **Microsoft.ML** is automatically installed for you by Model Builder.
+
+### Create a SparkSession
+
+1. Open the *Program.cs* file for **MLSparkModelML.ConsoleApp**. This file was autogenerated by Model Builder. Delete the `using` statements, the contents of the Main() method, and the `CreateSingleDataSample` region.
+
+1. Add the following additional `using` statements to the top of the *Program.cs*:
+
+   ```csharp
+   using System;
+   using System.Collections.Generic;
+   using Microsoft.ML;
+   using Microsoft.ML.Data;
+   using Microsoft.Spark.Sql;
+   using MLSparkModelML.Model;
+   ```
+
+1. Change the `DATA_FILEPATH` to the path of your *yelptest.csv*.
+
+1. Add the following code to your `Main` method to create a new `SparkSession`. The Spark Session is the entry point to programming Spark with the Dataset and DataFrame API.
+
+   ```csharp
+   SparkSession spark = SparkSession
+        .Builder()
+        .AppName(".NET for Apache Spark Sentiment Analysis")
+        .GetOrCreate();
+   ```
+
+   Calling the *spark* object created above allows you to access Spark and DataFrame functionality throughout your program.
+
+### Create a DataFrame and print to console
+
+Read in the Yelp review data from the *yelptest.csv* file as a `DataFrame`. Include `header` and `inferSchema` options.
+
+```csharp
+DataFrame df = spark
+    .ReadStream()
+    .Option("header", true)
+    .Option("inferSchema", true)
+    .Csv(DATA_FILEPATH);
+
+df.Show();
+```
+
+### Register a user-defined function
+
+You can use UDFs, *user-defined functions*, in Spark applications to do calculations and analysis on your data. In this tutorial, you use ML.NET with a UDF to evaluate each Yelp review.
+
+Add the following code to your `Main` method to register a UDF called `MLudf`.
+
+```csharp
+spark.Udf()
+    .Register<string, bool>("MLudf", predict);
```

Comment:
I think this UDF would need some more info/to call an actual `predict()` method?

In my sample, the UDF is defined as:

`
spark.Udf().Register<string, bool>(
                "MLudf",
                (text) => Sentiment(text, args[1]));
`

Reply:
The text is passed in when the UDF is called in the Spark SQL statement

Reply:
Yes I see that. But the Spark SQL statement takes in a string `ReviewText` as input- where does this definition of MLudf actually use a string? I see it's defined to take in a string parameter, but where does it actually define and use a string?

Reply:
First, we define predict(text). Then we register the UDF, which only requires the name, predict. Then we use MLudf(text) in Spark SQL, which passes in a string. 

Your function uses Sentiment(text, modelpath), but only the text is passed in the Spark SQL statement. I think that's why it needs to be added during UDF registration.

Reply:
Interesting, I didn't know that was possible! I thought `predict` would at least require parentheses to indicate it's a method. 

Thanks for the detailed explanation, and now I learned something new :)